### PR TITLE
Implement new dashboard search metadata

### PR DESCRIPTION
**Pull request by mijailr**
This reimplements the dashboard from scratch, starting with metadata search

Screenshot 2024-03-22 at 10 30 34 AM

### CODE REVIEW

File: dashboard-new/package.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "prettier --check .",
+    "lint:fix": "prettier --write ."
+  },
+  "dependencies": {
+    "@headlessui/react": "^1.7.18",
+    "@heroicons/react": "^2.1.1",
+    "littlehorse-client": "file://../sdk-js",
```
Comment (coltmcnealy-lh) [inline dashboard-new/package.json:15]: Ah, nice :smile: glad this is possible to do in JS.

File: dashboard-new/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# LH Dashboard
+
+## Development
```
Comment (coltmcnealy-lh) [inline dashboard-new/README.md:3]: This could use a bit of love (:

File: dashboard-new/src/app/getWhoami.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      },
+      tenants: ['default'],
+    }
+  }
+}
+
+const getTenants = ({ perTenantAcls, globalAcls }: Pick<Principal, 'globalAcls' | 'perTenantAcls'>): string[] => {
+  let tenants: string[] = []
+  if (globalAcls && hasDefaultAccess(globalAcls)) {
+    tenants = ['default']
```
Comment (coltmcnealy-lh) [inline dashboard-new/src/app/getWhoami.ts:35]: For this, we need to add an `rpc SearchTenants` and the Principal should have access to all tenants. CC @eduwercamacaro

File: dashboard-new/src/app/(authenticated)/workflow/[...workflow]/components/Details.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  ARCHIVED: 'bg-gray-200',
+  ACTIVE: 'bg-blue-200',
+  TERMINATING: 'bg-yellow-200',
+  UNRECOGNIZED: 'bg-red-200',
+}
+
+export const Details: FC<DetailsProps> = ({ id, status }) => {
+  return (
+    <div className="mb-4">
+      <span>Workflow</span>
```
Comment (coltmcnealy-lh) [inline dashboard-new/src/app/(authenticated)/workflow/[...workflow]/components/Details.tsx:22]: "Workflow" is ambiguous. Do you mean `WfSpec` or `WfRun`?

File: dashboard-new/src/app/(authenticated)/workflow/[...workflow]/components/Mutations.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
+import { VariableMutation } from 'littlehorse-client/dist/proto/common_wfspec'
+import { VariableValue } from 'littlehorse-client/dist/proto/variable'
+import { ThreadSpec } from 'littlehorse-client/dist/proto/wf_spec'
+import React, { FC, useMemo } from 'react'
+
+type Props = Pick<ThreadSpec, 'nodes'>
+export const Mutations: FC<Props> = ({ nodes }) => {
+  const mutations = useMemo(() => {
+    return Object.values(nodes).reduce<VariableMutation[]>((acc, node) => {
+      for (const mutation of node.outgoingEdges) {
```
Comment (coltmcnealy-lh) [inline dashboard-new/src/app/(authenticated)/workflow/[...workflow]/components/Mutations.tsx:10]: We should have a way to show _which outgoingEdge_ the mutation belongs to. The mutations don't belong to the `Node`; rather, they belong to each `edge`. Here is a `WfSpec` where this is important:

```
wf.doIfElse(wf.condition(foo, Comparator.LESS_THAN, bar), ifBody -> {
  ifBody.mutate(myOtherVar, VariableMutationType.ASSIGN, true);
}, elseBody -> {
  elseBody.mutate(myOtherVar, VariableMutationType.ASSIGN, valse);
});
```

File: dashboard-new/src/app/(authenticated)/workflow/[...workflow]/components/Variables.tsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  JSON_ARR: 'JSON Array',
+  DOUBLE: 'Double',
+  BOOL: 'Boolean',
+  STR: 'String',
+  INT: 'Integer',
+  BYTES: 'Bytes',
+  UNRECOGNIZED: 'Unrecognized',
+}
+
+const accessLevels: { [key in WfRunVariableAccessLevel]: string } = {
```
Comment (coltmcnealy-lh) [inline dashboard-new/src/app/(authenticated)/workflow/[...workflow]/components/Variables.tsx:18]: @eduwercamacaro this reminds me, we may need to think more about Variable Access Levels before `1.0.0`

File: dashboard-new/src/app/api/auth/signin/vertical-logo.svg
Comment (coltmcnealy-lh) [inline dashboard-new/src/app/api/auth/signin/vertical-logo.svg:1]: We should use our real logo I think

**Review by coltmcnealy-lh — APPROVED** [review]
Fantastic progress. Some things to clean up around tenant and principal management, also variable mutations, but overall it looks good!

File: dashboard-new/src/app/(authenticated)/workflow/[...workflow]/components/Details.tsx (hunk 4 above)
Comment (mijailr) [reply dashboard-new/src/app/(authenticated)/workflow/[...workflow]/components/Details.tsx:22]: In this context is WfSpec, I'm thinking on simplify the names,

WfSpec = Workflow
WfRun = Workflow Run

Or humanize

WfSpec = Workflow Spec
WfRun = Workflow Run

File: dashboard-new/src/app/getWhoami.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    const { perTenantAcls, globalAcls } = await client.whoami({})
+
+    const tenants = getTenants({ perTenantAcls, globalAcls })
+
+    return {
+      user: session.user,
+      tenants,
+    }
+  } else {
+    return {
```
[the remark this answers is not in the document]
Comment (mijailr) [reply dashboard-new/src/app/getWhoami.ts:22]: Understood, I'll fix this.

**Pull request merged**